packages/fiori/src/ShellBar.ts

```diff
@@ -699,29 +762,28 @@ class ShellBar extends UI5Element {
 	}
 
 	_handleActionsOverflow() {
-		const rightContainerRect = this.shadowRoot.querySelector(".ui5-shellbar-overflow-container-right").getBoundingClientRect();
+		const rightContainerRect = this.shadowRoot!.querySelector(".ui5-shellbar-overflow-container-right")!.getBoundingClientRect();
 		let overflowSelector = ".ui5-shellbar-button:not(.ui5-shellbar-overflow-button):not(.ui5-shellbar-invisible-button)";
 
 		if (this.showSearchField) {
 			overflowSelector += ",.ui5-shellbar-search-field";
 		}
 
-		const elementsToOverflow = this.shadowRoot.querySelectorAll(overflowSelector);
+		const elementsToOverflow = this.shadowRoot!.querySelectorAll<Button>(overflowSelector);
 		const isRTL = this.effectiveDir === "rtl";
 
-		let overflowCount = [].filter.call(elementsToOverflow, icon => {
-			const iconRect = icon.getBoundingClientRect();
+		const overflowCount = [...elementsToOverflow].filter(icon => {
+			const iconRect = (icon).getBoundingClientRect();
 
 			if (isRTL) {
 				return (iconRect.left + iconRect.width) > (rightContainerRect.left + rightContainerRect.width);
 			}
 
 			return iconRect.left < rightContainerRect.left;
 		});
+		const overFlowCountLength = !!overflowCount.length;
```

Comment:
The `overFlowCountLength` variable is used 
in  `_getAllItems(showOverflowButton: boolean)` 
as  `this._getAllItems(overFlowCountLength)`

Would not it be more readable and clean to rename `overFlowCountLength` to `showOverflow`
and use it `this._getAllItems(showOverflow)`